clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -3149,9 +3156,15 @@ struct ExpandPackedTypeConstraints
 
     assert(SemaRef.ArgumentPackSubstitutionIndex != -1);
 
+    TemplateArgument Arg = TemplateArgs(T->getDepth(), T->getIndex());
+
+    std::optional<unsigned> PackIndex;
+    if (Arg.getKind() == TemplateArgument::Pack)
+      PackIndex = Arg.pack_size() - 1 - SemaRef.ArgumentPackSubstitutionIndex;
```

Comment:
Can't we use `getPackIndex` here:

https://github.com/zyn0217/llvm-project/blob/734f00daf65c079c63c8c338122c02ce56130df6/clang/lib/Sema/SemaTemplateInstantiate.cpp#L1385-L1390

Reply:
These index transformations also come up in other places, where some other index is used, instead of the current Sema pack subst index. That would be worth refactoring, also consolidating asserts.